File: src/kubernetes/events.ts
// convert to to using the kubernetes sdk

export async function saveEvents(namespace: string) {
  try {
    const events = new Deno.Command('kubectl', { args: ['get', 'events', '-n', namespace, '--sort-by=.metadata.creationTimestamp'] });
    const output = await events.output();
    await Deno.writeTextFile(`${dirPath}/Events.txt`, new TextDecoder().decode(output.stdout));
  } catch (error) {
    console.error(`Error saving events:`, error);
  }
}

File: internal/k8s/events.go
package k8s

import (
	"bytes"
	"fmt"
	"os/exec"
)

func Events(namespace string) (string, error) {
	cmd := exec.Command("kubectl", "get", "events", "-n", namespace, "--sort-by=.metadata.creationTimestamp")
	var out bytes.Buffer
	var stderr bytes.Buffer
	cmd.Stdout = &out
	cmd.Stderr = &stderr
	err := cmd.Run()
	if err != nil {
		return "", fmt.Errorf("error getting k8s events: %v: %s", err, stderr.String())
	}
	return out.String(), nil
}
